src/lexer/lexer.py

```diff
@@ -400,6 +454,22 @@ def _seek(self, to_seek: str|list[str], before: bool = False, multi_line_count:
         '''
         pass
 
+    def _check_prev_token(self, to_check: TokenType | list[TokenType]):
+        """
+        Checks prev token if it has the token type/s passed. Must be on the same line
+        """
+
+        # Convert input to list
+        if isinstance(to_check, TokenType):
```

Comment:
please provide another validation of input
```
elif not isinstance(to_check, list) and all(isinstance(token, TokenType) for token in to_check):
    raise ValueError(f"{to_check} is not a valid parameter. Please pass a single TokenType or a list of TokenTypes")
```

Reply:
now that I think about it, we don't really need to change this since this is a private function. changing it would lead to QOL when debugging (example: typo) but we don't really need it

I'll change my review to approved! *but feel free to change it if ever*